binding/fortran/source/calculate.f90

```diff
@@ -184,10 +193,12 @@ subroutine assignExpression(this, other)
         class(Expression), intent(inout) :: this
         type(Expression), intent(in) :: other
 
+        type(LibraryTemplate), pointer :: Calculate
         procedure(newExpressionWrapper), pointer :: new
 
         if (other%init_number == MAGIC_NUMBER) then
-            call c_f_procpointer(CalculateLibrary%newExpression, new)
+            call c_f_pointer(libraryReference(), Calculate)
```

Comment:
**Fortran** compilers ignore extern globals as a requirement, so the linker didn't link against them returning arbitrary memory locations. The solution was to define a function that returns a reference to the library, as the linker can't link against undefined methods.